plan/plan.go

```diff
@@ -163,6 +164,8 @@ type LogicalPlan interface {
 	// deriveStats derives statistic info between plans.
 	deriveStats() (*statsInfo, error)
 
+	cardinality() float64
```

Comment:
Can we add `getStats` instead?
Then use the `count` to determine the outer table.